eZ/Publish/Core/Persistence/Legacy/Content/Handler.php

```diff
@@ -554,6 +554,11 @@ public function deleteVersion( $contentId, $versionNo )
         $this->contentGateway->deleteRelations( $contentId, $versionNo );
         $this->contentGateway->deleteVersions( $contentId, $versionNo );
         $this->contentGateway->deleteNames( $contentId, $versionNo );
+
+        if ( $this->contentGateway->getLastVersionNumber( $contentId ) == 0 )
```

Comment:
Indeed, the function would return null except for the `(int)` typecast.
This comparison could even be done against `null`, but I see how it looks "kind of weird".

Thanks for the feedback, I will look into the tests too.